views/utils.py

```diff
@@ -112,10 +86,44 @@ def _load_challenges() -> dict[ChallengeName, Challenge]:
         return challenges
 
     @staticmethod
-    def _type_check_with_mypy(code) -> TypeCheckResult:
+    def _type_check_with_mypy(code: str) -> TypeCheckResult:
+        buffer = io.StringIO(code)
+        tokens = list(tokenize.generate_tokens(buffer.readline))
+        # Find all lines that are followed by a comment # expect-type-error
+        expect_error_lines = [
+            token.start[0]
+            for token in tokens
+            if token.type == tokenize.COMMENT
+            and token.string[1:].strip() == EXPECT_ERROR_COMMENT
+        ]
         raw_result = api.run(["--config-file", str(MYPY_CONFIG), "-c", code])
+        error_lines: list[str] = []
+
+        for line in raw_result[0].splitlines():
+            m = re.match(MYPY_MESSAGE_REGEX, line)
```

Comment:
nit: can use `:=`

Reply:
The m is still being used by the code outside of if-statement, it's okay to keep it.